analysis/tools/ccsh/src/main/kotlin/de/maibornwolff/codecharta/tools/ccsh/parser/ParserService.kt

```diff
@@ -0,0 +1,60 @@
+package de.maibornwolff.codecharta.tools.ccsh.parser
+
+import com.github.kinquirer.KInquirer
+import com.github.kinquirer.components.promptList
+import de.maibornwolff.codecharta.filter.edgefilter.EdgeFilter
+import picocli.CommandLine
+
+class ParserService {
+    companion object {
+        fun selectParser(commandLine: CommandLine): String {
+            val selectedParser: String = KInquirer.promptList(
+                message = "Which parser do you want to execute?",
+                choices = getParserNamesWithDescription(commandLine)
+            )
+
+            return extractParserName(selectedParser)
+        }
+
+        fun executeSelectedParser(selectedParser: String) {
+            val args = emptyArray<String>()
```

Comment:
Yes, it is being used on line 26. The parsers main method needs to be called with an empty array, so instead of passing an empty array to each, we declare it once. For now, there is only the edgefilter interactive dialog implemented. But the goal is to implement it in all parser and get rid of the "print("Not implemented yet")".